tardis/model/base.py

```diff
@@ -383,3 +386,124 @@ def from_config(cls, config):
                    v_boundary_inner=structure.get('v_inner_boundary', None),
                    v_boundary_outer=structure.get('v_outer_boundary', None),
                    electron_densities=electron_densities)
+    @classmethod
+    def from_csvy(cls, config):
```

Comment:
horrible written function 😉 - but I know that this is not from you. Should we clean it up - or defer to later?

Reply:
I think let's defer that to later, and a different PR.  But I agree that it is not that well organized.  I will think about how to improve this.